components/builder-api/src/server/resources/jobs.rs

```diff
@@ -122,14 +125,54 @@ fn get_rdeps(req: HttpRequest,
     match route_message::<jobsrv::JobGraphPackageReverseDependenciesGet,
                         jobsrv::JobGraphPackageReverseDependencies>(&req, &rdeps_get)
     {
-        Ok(rdeps) => HttpResponse::Ok().json(rdeps),
+        Ok(rdeps) => {
+            let filtered = match filtered_rdeps(&req, rdeps) {
+                Ok(f) => f,
+                Err(err) => return err.into(),
+            };
+            HttpResponse::Ok().json(filtered)
+        }
         Err(err) => {
             debug!("{}", err);
             err.into()
         }
     }
 }
 
+fn filtered_rdeps(req: &HttpRequest,
+                  rdeps: jobsrv::JobGraphPackageReverseDependencies)
+                  -> Result<jobsrv::JobGraphPackageReverseDependencies> {
+    let mut new_rdeps = jobsrv::JobGraphPackageReverseDependencies::new();
+    new_rdeps.set_origin(rdeps.get_origin().to_string());
+    new_rdeps.set_name(rdeps.get_name().to_string());
+
+    let mut origin_map = HashMap::new();
+    let mut short_deps = RepeatedField::new();
+
+    for rdep in rdeps.get_rdeps() {
+        let ident = OriginPackageIdent::from_str(rdep)?;
+        let origin_name = ident.get_origin();
+        let pv = if !origin_map.contains_key(origin_name) {
+            let conn = req_state(req).db.get_conn().map_err(Error::DbError)?;
+            let origin = Origin::get(&origin_name, &*conn)?;
+            origin_map.insert(origin_name.to_owned(),
+                              origin.default_package_visibility.clone());
+            origin.default_package_visibility
+        } else {
+            origin_map.get(origin_name).unwrap().clone()
+        };
+        if pv != PackageVisibility::Public && authorize_session(req, Some(&origin_name)).is_err() {
+            debug!("Skipping unauthorized non-public origin package: {}", rdep);
+            continue; // Skip any unauthorized origin packages
```

Comment:
I'm pondering if quietly skipping the data leaves the user in an entirely favorable situation. Perhaps they need to know that there is data missing so that they can make an adjustment?  I'm almost thinking that a "breaking" response such as a HTTP 403 or inserting "<the_origin_name>/unauthorized_for_private_visbility" as the rdep might be more helpful for anyone attempting to actually get value from the data?  Otherwise, they are left assuming that they got a comprehensive list when they might not have if unauthorized for one or more of the rdeps and likely will attempt to use the list in good faith.

Thoughts?

Reply:
Good q - atm the rdeps API is only used as an advisory when you do a 'hab bldr job start', so the exposure is quite limited.  I'm going to follow up with some changes to that code to start sending in an auth token, and can add a brief warning there that private packages will not be shown.

Reply:
I think that sounds like a great plan. 👍